Trees/deserialize_serialize_binary_tree.py

```diff
@@ -0,0 +1,37 @@
+class Solution():
+  '''
+  this problem shows how to convert a binary tree to serialized form which can make it easier to transfer/write to other formats.
+  The method to deserialize is given as well
+  '''
+  def serialize(self, root) :
+    nodes = [] # the 
+
+    def dfs(node):
+      if not node:
+        return "N"
+      #preorder addition of nodes to our list
+      nodes.append(node.value)
+      nodes.append(dfs(node.left)) # I need to draw a picture that shows the nodes getting added to the list cause I'm confused
```

Comment:
you are confused because this is wrong. the appending already happens on the previous line. add a comment to call these the break down of the recursive problem